sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -236,19 +245,13 @@ case class And(left: Expression, right: Expression) extends BinaryOperator with
 
   override def eval(input: InternalRow): Any = {
     val input1 = left.eval(input)
```

Comment:
For `And` and `Or`, I realize that we probably cannot do the short circuit optimization in runtime, as it's not equivalent for case like `A AND B` and `B AND A`.